cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedProduct.js

```diff
@@ -374,17 +384,23 @@ function algoliaLocalizedProduct(parameters) {
             } else {
                 var config = algoliaProductConfig.attributeConfig_v2[attributeName];
 
-                // if there is no config for the attribute, but it's a custom or activeData attribute, we assume some defaults
-                if (!config && (attributeName.indexOf('custom.') > -1 || attributeName.indexOf('activeData.') > -1)) {
-                    // if the attribute is a custom attribute, we assume it's localized, you can override this behavior by adding a config for the attribute
-                    config = {
-                        attribute: attributeName,
-                        localized: true,
-                        variantAttribute: true
-                    };
-                }
+                if (empty(config)) {
+                    config = jobHelper.getCustomAttributeConfig(attributeName);
 
-                if (!empty(config)) {
+                    if (attributeNameObj.length > 1) {
+                        var tempObj;
+                        if (this[parentAttribute]) {
+                            tempObj = this[parentAttribute];
+                            tempObj[subAttribute] = ObjectHelper.getAttributeValue(product, config.attribute);
+                        } else {
+                            tempObj = {};
+                            tempObj[subAttribute] = ObjectHelper.getAttributeValue(product, config.attribute);
+                        }
+                        this[parentAttribute] = tempObj;
```

Comment:
I think this can be simplified like that:
```suggestion
                        if (!this[parentAttribute]) {
                            this[parentAttribute] = {};
                        }
                        this[parentAttribute][subAttribute] = ObjectHelper.getAttributeValue(product, config.attribute);
```